File: src/spice2x/api/resources/dart/spiceapi/src/wrappers/memory.dart
part of spiceapi;

Future<void> memoryWrite(Connection con,
    String dllName, String data, int offset) {
  var req = Request("memory", "write");
  req.addParam(dllName);
  req.addParam(data);
  req.addParam(offset);
  return con.request(req);
}

Future<String> memoryRead(Connection con,
    String dllName, int offset, int size) {
  var req = Request("memory", "read");
  req.addParam(dllName);
  req.addParam(offset);
  req.addParam(size);
  return con.request(req).then((res) {
    return res.getData()[0];
  });
}

Future<int> memorySignature(Connection con,
    String dllName, String signature, String replacement,
    int offset, int usage) {
  var req = Request("memory", "signature");
  req.addParam(dllName);
  req.addParam(signature);
  req.addParam(replacement);
  req.addParam(offset);
  req.addParam(usage);
  return con.request(req).then((res) {
    return res.getData()[0];
  });
}

File: src/spice2x/api/resources/python/spiceapi/memory.py
from .connection import Connection
from .request import Request


def memory_write(con: Connection, dll_name: str, data: str, offset: int):
    req = Request("memory", "write")
    req.add_param(dll_name)
    req.add_param(data)
    req.add_param(offset)
    con.request(req)


def memory_read(con: Connection, dll_name: str, offset: int, size: int):
    req = Request("memory", "read")
    req.add_param(dll_name)
    req.add_param(offset)
    req.add_param(size)
    res = con.request(req)
    return res.get_data()[0]


def memory_signature(con: Connection, dll_name: str, signature: str,
                     replacement: str, offset: int, usage: int):
    req = Request("memory", "signature")
    req.add_param(dll_name)
    req.add_param(signature)
    req.add_param(replacement)
    req.add_param(offset)
    req.add_param(usage)
    res = con.request(req)
    return res.get_data()[0]
